#! /bin/sh /usr/share/dpatch/dpatch-run
## 12_mips_atomic_ops.dpatch by Thiemo Seufer <ths@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fix compilation for Linux MIPS-I code.

@DPATCH@
diff -urNad qt4-x11-4.1.0~/src/corelib/arch/mips/qatomic32.s qt4-x11-4.1.0/src/corelib/arch/mips/qatomic32.s
--- qt4-x11-4.1.0~/src/corelib/arch/mips/qatomic32.s	2005-12-16 06:52:19.000000000 -0500
+++ qt4-x11-4.1.0/src/corelib/arch/mips/qatomic32.s	2006-01-07 10:47:37.000000000 -0500
@@ -6,10 +6,12 @@
 	.globl	q_atomic_test_and_set_int
 	.ent	q_atomic_test_and_set_int
 q_atomic_test_and_set_int:
+	.set mips2
 1:	ll   $8,0($4)
 	bne  $8,$5,2f
 	move $2,$6
 	sc   $2,0($4)
+	.set mips0
 	beqz $2,1b
 	nop
 	jr   $31
@@ -21,10 +23,12 @@
 	.globl	q_atomic_test_and_set_acquire_int
 	.ent	q_atomic_test_and_set_acquire_int
 q_atomic_test_and_set_acquire_int:
+	.set mips2
 1:	ll   $8,0($4)
 	bne  $8,$5,2f
 	move $2,$6
 	sc   $2,0($4)
+	.set mips0
 	beqz $2,1b
 	nop
 	jr   $31
@@ -38,10 +42,12 @@
 	.ent	q_atomic_test_and_set_release_int
 q_atomic_test_and_set_release_int:
 	sync
+	.set mips2
 1:	ll   $8,0($4)
 	bne  $8,$5,2f
 	move $2,$6
 	sc   $2,0($4)
+	.set mips0
 	beqz $2,1b
 	nop
 	jr   $31
@@ -53,10 +59,12 @@
 	.globl	q_atomic_test_and_set_ptr
 	.ent	q_atomic_test_and_set_ptr
 q_atomic_test_and_set_ptr:
+	.set mips2
 1:	ll   $8,0($4)
 	bne  $8,$5,2f
 	move $2,$6
 	sc   $2,0($4)
+	.set mips0
 	beqz $2,1b
 	nop
 	jr   $31
